adelie/sklearn.py

```diff
@@ -249,6 +251,84 @@ def _validate_params(self):
         ]:
             raise ValueError(f"Unknown family: {self.family}")
 
+    def plot_validation_curve(self, X_val, y_val):
+        """
+        Plot validation curve showing R^2 and active set size at each lambda.
+
+        Parameters
+        ----------
+        X_val : array-like
+            Validation feature matrix.
+        y_val : array-like
+            Validation response vector.
+        """
+        import matplotlib.pyplot as plt
+        import matplotlib.ticker as mticker
+        from sklearn.metrics import r2_score
+        import numpy as np
+
+        # Compute predictions for each model along the path
+        predictions = self.predict(X_val)
+        # If predictions is 2D (n_lambdas, n_samples), axis=1 is correct
+        # If predictions is 1D, wrap in a 2D array for apply_along_axis
+        if predictions.ndim == 1:
+            predictions = predictions[np.newaxis, :]
+        r2vec = np.apply_along_axis(lambda yhat: r2_score(y_val, yhat), axis=1, arr=predictions)
+
+        lam_path = self.lambda_
+        beta_0 = self.intercept_
+        # Handle sparse or dense coef_
+        betas = self.coef_.toarray() if hasattr(self.coef_, "toarray") else np.asarray(self.coef_)
```

Comment:
I would prefer to perform this check by doing `isinstance`. It's more explicit about what we expect. It'll either be `csr_matrix` or `np.ndarray` right?